packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -119,6 +120,7 @@ class ScriptTransformer {
     if (transformerCacheKey != null) {
       return createHash('sha1')
         .update(transformerCacheKey)
+        .update(transformOptions.configString)
```

Comment:
This fixes the issue where different transformers end up using each other's cache directory if they happen to return the same `transformerCacheKey`

Reply:
I still think there's something fishy going on here though. I'm able to break the existing tests, without making any changes, simply by changing a few `getCacheKey` returns, in ways that don't make sense.

Reply:
could you expand on your last comment? In what way?

Reply:
@SimenB Checkout this PR https://github.com/facebook/jest/pull/13985/files it demonstrates the failing test.
In that test we can see the the 4th transformer is reusing the project cache from a different transformer.